cmd/controller/main.go

```diff
@@ -4,18 +4,72 @@
 package main
 
 import (
+	"context"
 	"flag"
-	"os"
-	"time"
-
+	"github.com/go-logr/logr"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/leaderelection"
+	"k8s.io/client-go/tools/leaderelection/resourcelock"
 	"k8s.io/klog/v2"
+	"os"
 	logf "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	"time"
 )
 
 // Version of kapp-controller is set via ldflags at build-time from the most recent git tag; see hack/build.sh
 var Version = "develop"
 
+var (
+	client *clientset.Clientset
+)
+
+func getNewLock(lockname, podname, namespace string) *resourcelock.LeaseLock {
```

Comment:
Early Feedback: Can we move all the leader election code to a different file / module? Main ought to be as simple as possible (just start the app)

Reply:
okay will look into this